File: day8.py
string_literals = 0
chrs_in_memory = 0

with open('./inputs/day8.txt', 'r') as f:
  for line in f:
    string_literals += len(line) - 1
    chrs_in_memory += len(eval(line))

print 'string_literals: ' + str(string_literals)
print 'chrs_in_memory: ' + str(chrs_in_memory)
print string_literals - chrs_in_memory

acc = 0

with open('./inputs/day8.txt', 'r') as f:
  for line in f:
    acc += 2 + line.count('\\') + line.count('"')

print acc

File: 2021/js/day_8.js
const path = require('path');
const fs = require('fs');

const input = fs.readFileSync(path.join(__dirname, '../', 'inputs', 'day8.txt'), { encoding: 'utf-8' })
  .split('\n').reduce((acc, line) => acc.concat([line.split(' | ')[1].split(' ')]), []);

function countTokens(input) {
  return input.flat().reduce((acc, token) => acc + Number([2, 3, 4, 7].includes(token.length)), 0);
}

console.log(countTokens(input))